src/react/account/AccountUpdateUser.tsx

```diff
@@ -0,0 +1,191 @@
+// @flow
+import FormContainer, * as F from '../ui-elements/FormLayout';
+import React, { useRef } from 'react';
+import { clearError, handleErrorMessage, networkEnd } from '../actions';
+import { useDispatch, useSelector } from 'react-redux';
+import type { AppState } from '../../types/state';
+import { Banner } from '@scality/core-ui';
+import { Button } from '@scality/core-ui/dist/next';
+import Joi from '@hapi/joi';
+import { joiResolver } from '@hookform/resolvers';
+import { useForm } from 'react-hook-form';
+import { useOutsideClick } from '../utils/hooks';
+import { useIAMClient } from '../IAMProvider';
+import { queryClient } from '../App';
+import { useMutation } from 'react-query';
+import { useHistory, useParams } from 'react-router-dom';
+const regexpName = /^[\w+=,.@ -]+$/;
+const schema = Joi.object({
+  name: Joi.string()
+    .label('Name')
+    .required()
+    .min(2)
+    .max(64)
+    .regex(regexpName)
+    .message('Invalid Name'),
+});
+
+const AccountUpdateUser = () => {
+  const dispatch = useDispatch();
+  const history = useHistory();
+  const IAMClient = useIAMClient();
+  const { IAMUserName } = useParams();
+  const { register, handleSubmit, errors } = useForm({
+    resolver: joiResolver(schema),
+  });
+  const { accountName } = useParams();
```

Comment:
I think you have something like : 
```suggestion
  const { IAMUserName, accountName } = useParams<{ IAMUserName: string, accountName: string }>();
  const { register, handleSubmit, errors } = useForm({
    resolver: joiResolver(schema),
  });
```